### DISCUSSION TITLE
Nextcloud Talk High performance backend issue

### DISCUSSION THREAD

**Discussion by IgnaceMenace** [nextcloud/all-in-one#1662] [Questions] [answered]
Hi,
(I've been redirected here from reddit)
I have a running Nextcloud AIO (https://raw.githubusercontent.com/IgnaceMenace/server/main/nextcloud/docker-compose.yml) on my home server and it works well excepted Nextcloud Talk which I have some issues with.

First, Portainer says Nextcloud-AIO-Talk is unhealthy

Second, I have a 502 error for the high performance backend

Third, I can't send message from the web client but I can react with emojis... and I have a notification saying "Failed to establish signaling connection"

Even though, Nextcloud talk works perfectly fine from the Android app.
I'm running it behind a reverse proxy and I have forwarded port 3478 in my router to my server.

Anybody knows what I can do to fix that ?

Thanks.
![image](https://user-images.githubusercontent.com/74070019/210366069-6ae2f4ea-148b-4a39-b927-927d54ad961d.png)

**Comment by Zoey2936** [comment]
Could you please send the logs of the nextcloud-aio-talk container please?
And please try to use the official [compose file](https://github.com/nextcloud/all-in-one/blob/main/docker-compose.yml), since the file provided by you contains things that make no sense at all.

**Reply by Zoey2936** [reply]
And also your reverse proxy configuration

**Reply by IgnaceMenace** [reply]
Thank you for your answer, I have deleted my containers and volume then compose it with the official file and uncommented lines related to reverse proxy usage.
Here is the logs  : main.go:228: Could not initialize janus MCU (dial tcp 127.0.0.1:8188: connect: connection refused) will retry in 16s
(Its the exact same line multiple time so I just pasted it once)

I didn't knew about the log until you asked but then I thought I could paste it in my search engine and I found that a guy had the exact same problem a month ago and on the same server operating server.
The issue doesn't seem to be solved even if Szaimen tried to answer.
https://help.nextcloud.com/t/talk-not-able-to-start-error-talking-to-server/152286/4

**Reply by Zoey2936** [reply]
I am not sure, but maybe this could be an Issue on SE-Linux? I'm not sure if fedora uses SE-Linux, but I've read a lot that docker and SE-Linux can make problems

**Reply by IgnaceMenace** [reply]
Actually the logs look like that after a restart (sorry for the first logs)
Debug/log level is 4
Debug/log timestamps are disabled
Debug/log colors are disabled
Adding 'vmnet' to the ICE ignore list...
Using 172.17.0.4 as local IP...
Token based authentication disabled
Initializing recorder code
Initializing ICE stuff (Full mode, ICE-TCP candidates disabled, full-trickle, IPv6 support disabled)
STUN server to use: 127.0.0.1:3478
  >> 127.0.0.1:3478 (IPv4)
Testing STUN server: message is of 20 bytes
  >> Our public address is 127.0.0.1
TURN server to use: 127.0.0.1:3478 (udp)
TURN REST API backend: (disabled)
Crypto: OpenSSL >= 1.1.0
No cert/key specified, autogenerating some...
Fingerprint of our certificate: 78:61:2E:20:5D:15:7D:5E:3B:1A:3D:A8:BF:E8:3B:85:2D:E7:F6:4A:F9:0C:28:73:6A:BB:9D:D1:30:1D:71:BB
Event handlers support disabled
Plugins folder: /usr/lib/x86_64-linux-gnu/janus/plugins
Sessions watchdog started
Joining Janus requests handler thread
Loading plugin 'libjanus_audiobridge.so'...
JANUS AudioBridge plugin initialized!
Loading plugin 'libjanus_duktape.so'...
[echotest.js]  Loading script...
[echotest.js]  Modules folder: /usr/share/janus/duktape
[echotest.js]  Loading module: janus-sdp
[echotest.js]  Module loaded
[echotest.js]  Script loaded
[echotest.js]  Initializing...
[echotest.js]  Initialized
Janus JavaScript plugin (Duktape) initialized!
Loading plugin 'libjanus_echotest.so'...
JANUS EchoTest plugin initialized!
Loading plugin 'libjanus_lua.so'...
[echotest.lua] Loading...
[echotest.lua] Loaded
[echotest.lua] Initializing...
[echotest.lua] Initialized
Janus Lua plugin initialized!
Loading plugin 'libjanus_nosip.so'...
JANUS NoSIP plugin initialized!
Loading plugin 'libjanus_recordplay.so'...
JANUS Record&Play plugin initialized!
Loading plugin 'libjanus_sip.so'...
JANUS SIP plugin initialized!
Loading plugin 'libjanus_streaming.so'...
[ERR] [plugins/janus_streaming.c:janus_streaming_init:2492] Can't add 'live' mountpoint, no such file '/usr/share/janus/streams/radio.alaw'...
[ERR] [plugins/janus_streaming.c:janus_streaming_init:2566] Can't add 'ondemand' mountpoint, no such file '/usr/share/janus/streams/music.mulaw'...
JANUS Streaming plugin initialized!
Loading plugin 'libjanus_textroom.so'...
JANUS TextRoom plugin initialized!
Loading plugin 'libjanus_videocall.so'...
JANUS VideoCall plugin initialized!
Loading plugin 'libjanus_videoroom.so'...
JANUS VideoRoom plugin initialized!
Loading plugin 'libjanus_voicemail.so'...
JANUS VoiceMail plugin initialized!
Transport plugins folder: /usr/lib/x86_64-linux-gnu/janus/transports
Loading transport plugin 'libjanus_http.so'...
HTTP transport timer started
HTTP webserver started (port 8088, /janus path listener)...
JANUS REST (HTTP/HTTPS) transport plugin initialized!
Loading transport plugin 'libjanus_mqtt.so'...
MQTT SSL support disabled
[WARN] MQTT support disabled for both Janus and Admin API, giving up
JANUS MQTT transport plugin destroyed!
[WARN] The 'janus.transport.mqtt' plugin could not be initialized
Loading transport plugin 'libjanus_nanomsg.so'...
JANUS Nanomsg transport plugin initialized!
Loading transport plugin 'libjanus_pfunix.so'...
Nanomsg thread started
[WARN] No Unix Sockets server started, giving up...
[WARN] The 'janus.transport.pfunix' plugin could not be initialized
Loading transport plugin 'libjanus_rabbitmq.so'...
RabbitMQ SSL support disabled
[WARN] RabbitMQ support disabled for both Janus and Admin API, giving up
[WARN] The 'janus.transport.rabbitmq' plugin could not be initialized
Loading transport plugin 'libjanus_websockets.so'...
libwebsockets logging: 0
[ERR] [transports/janus_websockets.c:janus_websockets_init:737] Error creating libwebsockets context...
[WARN] The 'janus.transport.websockets' plugin could not be initialized
main.go:228: Could not initialize janus MCU (dial tcp 127.0.0.1:8188: connect: connection refused) will retry in 8s
...

**Reply by IgnaceMenace** [reply]
I will try to fix it and if i find an answer I ll post it here thx for your help
I have also heard of SeLinux so I'll start with that

**Reply by Zoey2936** [reply]
Can you try the following commands and send the output?
`docker run --rm -it -e NC_DOMAIN=h -e TURN_SECRET=h -e SIGNALING_SECRET=h -e JANUS_API_KEY=h nextcloud/aio-talk`
`docker run --rm -it -e NC_DOMAIN=h -e TURN_SECRET=h -e SIGNALING_SECRET=h -e JANUS_API_KEY=h --privileged nextcloud/aio-talk`

**Reply by IgnaceMenace** [reply]
[notPriviledged.log](https://github.com/nextcloud/all-in-one/files/10337841/notPriviledged.log)
[priviledged.log](https://github.com/nextcloud/all-in-one/files/10337842/priviledged.log)

**Comment by szaimen** [comment]
Hi, I googled a bit and is it possible that ulimit for the docker service on systemd is set to infinity? Can you try to change this to a limited but still high value? This may resolve you issue with Janus.

I found this: https://github.com/canyanio/janus-gateway-docker/issues/22

> This happens if the ulimit is set to infinity on systemd for the docker service (or cgroups/whatever you are using) and because of how the code is written [here](https://github.com/warmcat/libwebsockets/blob/6ed3d8a9dd68bb1d762a08312dfe4a65aed80922/lib/core/context.c#L562-L566).

**Reply by olovm** [reply]
I have the same problem as described here, running on a Fedora 36 host.

Do you have any pointers on how I can change these values when starting the Nexcloud aio using the instructions from: https://github.com/nextcloud/all-in-one

I tried to change the docker run command by adding: 
--ulimit nofile=200000:200000
but that only changed the ulimit in the mastercontainer, not in talk (checked by entering the running container and calling ulimit -a).

**Reply by szaimen** [reply]
Hi, it is currently not possible to set this value for specific containers. That is why I asked if you could try to limit it for the docker service on systemd again?

**Reply by szaimen** [reply]
However for a test you could try this one and please post your logs here afterwards: 
```
docker run --rm -it -e NC_DOMAIN=h -e TURN_SECRET=h -e SIGNALING_SECRET=h -e JANUS_API_KEY=h --ulimit nofile=200000:200000 nextcloud/aio-talk
```

**Reply by IgnaceMenace** [reply]
Hi thanks for your help but unfortunately it doesn't change the error or the logs, and I restarted everything...
I don't want to  waste to much of your time with that
A quick "fix" to be able to send msg from the web interface anyway is to delete the high performance backend from the  administration settings but well it is of course not the best idea
![image](https://user-images.githubusercontent.com/74070019/210572324-e903e402-1da8-43a1-9c0f-16862a2bfed7.png)

**Reply by szaimen** [reply]
> However for a test you could try this one and please post your logs here afterwards:
> 
> ```
> docker run --rm -it -e NC_DOMAIN=h -e TURN_SECRET=h -e SIGNALING_SECRET=h -e JANUS_API_KEY=h --ulimit nofile=200000:200000 nextcloud/aio-talk
> ```

can you send the logs of this command? It is obvious that it does not solve the talk problem directly as this is just a test with an additional container.

**Reply by IgnaceMenace** [reply]
The test container is healthy and running
[_compassionate_keldysh_logs.txt](https://github.com/nextcloud/all-in-one/files/10344878/_compassionate_keldysh_logs.txt)

**Reply by szaimen** [reply]
> The test container is healthy and running
> [_compassionate_keldysh_logs.txt](https://github.com/nextcloud/all-in-one/files/10344878/_compassionate_keldysh_logs.txt)

Interesting, thanks. Then I guess we I will need to look into how to add a ulimit to the container.

**Reply by IgnaceMenace** [reply]
Ok Thank you
Have a nice day

**Reply by szaimen** [reply]
This is what it needs

```
"HostConfig": {
            "Ulimits": [
                {
                    "Name": "nofile",
                    "Hard": 200000,
                    "Soft": 200000
                }
            ],
```

**Reply by IgnaceMenace** [reply]
Thank you very much but sorry I don't understand where I have to write that

**Reply by szaimen** [reply]
This was only for me. The PR https://github.com/nextcloud/all-in-one/pull/1671 is already merged and I will let you know when it is available in the beta channel

**Reply by IgnaceMenace** [reply]
alright well done

**Comment by szaimen** [comment, marked as answer]
This is now released with v4.1.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by IgnaceMenace** [reply]
It works perfectly fine !

**Reply by szaimen** [reply]
great! good to hear!
